### PR TITLE
Support attach metadata in transaction

### PR DESCRIPTION
**Pull request by h2physics**

### CODE REVIEW

File: txbuilder/txbuilder.go [hunk 1]
```diff
@@ -279,6 +280,12 @@ func SetValidRangeTo(to int64) Option {
 	}
 }
 
+func SetJsonMetadata(path string) Option {
```
Comment (longngn) [inline txbuilder/txbuilder.go:283]: func SetMetadata(json string)

File: txbuilder/txbuilder.go [hunk 2]
```diff
@@ -86,6 +86,7 @@ type TxBuilder struct {
 	Collaterals         []TxInput
 	ValidRangeFrom      *int64
 	ValidRangeTo        *int64
+	JsonMetadataPath    string
```
Comment (longngn) [inline txbuilder/txbuilder.go:89]: JsonMetadata

**Review by longngn — CHANGES_REQUESTED** [review]

File: txbuilder/txbuilder.go (hunk 1 above)
Comment (h2physics) [reply txbuilder/txbuilder.go:283]: should be detailed naming here because having 2 ways to attach metadata (json & cbor). Maybe `cbor` will be used later

Comment (longngn) [reply txbuilder/txbuilder.go]: then `SetJSONMetadata`

File: txbuilder/txbuilder.go [hunk 3]
```diff
@@ -86,7 +86,7 @@ type TxBuilder struct {
 	Collaterals         []TxInput
 	ValidRangeFrom      *int64
 	ValidRangeTo        *int64
-	JsonMetadataPath    string
+	JsonMetadata        string
```
Comment (longngn) [inline txbuilder/txbuilder.go:89]: JSONMetadata
I forgot Go's convention lol

**Pull request merged**